.github/workflows/deploy.yml

```diff
@@ -0,0 +1,54 @@
+name: Generate Documentation and deploy to GitHub pages
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+permissions:
+  pages: write
+  id-token: write
+
+concurrency:
+  group: pages
+  cancel-in-progress: true
+
+jobs:
+  deploy:
+    name: Build and Deploy 
+    runs-on: ubuntu-latest
+    environment:
+      name: github-pages
+      url: ${{ steps.deploy.outputs.page_url }}
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      - name: Setup Python
+        id: python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install Dependencies
+        id: dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Build Documentation
+        id: build
+        run: mkdocs build
+
+      - name: Upload Artifact
+        id: upload
+        uses: actions/upload-pages-artifact@v1
+        with:
+          path: ./site/
+          
+      - name: Deploy
+        id: deploy
+        uses: actions/deploy-pages@v1
```

Comment:
More of a note to self, but this will deploy to a different URL, breaking existing links. I need to do some redirects and stuff to make this all work.

Reply:
To clarify, this will deploy without relying on a branch. In the repository's settings for pages it should be set to GitHub Actions if you weren't already using that. The URL should still be https://miguelguthridge.github.io/FL-Studio-API-Docs/ though.

Reply:
Neat, if that's correct, this should be a huge improvement on the massively convoluted system I was using before (literally using API keys to copy stuff from one repo to another that gets statically hosted). I can't see any definition in the file though. Are you sure it won't just deploy to whatever the repo name is?

Reply:
I'll add comments to the workflow file, to make it easier for both of us to understand what's going on.